vtex-sites/nextjs.store
# fix: `view_item_list` event being fired without products

## What's the purpose of this pull request?

Fixed an issue where the `view_item_list` analytics event is being fired with an empty list of `items` when viewing the list of products on the PLPs. We also configured the `ProductShelf` and `ProductTiles` components to fire this event because they render a list of products (e.g. "Most Wanted" and "Just Arrived now!" sections on the Home page).

## How to test it?

Integration tests were added, and they assert the event is fired as expected.

You can check out the `dataLayer` events being fired on https://preview-242--nextjs.preview.vtex.app/ following this guide [See events pushed to `dataLayer`](https://www.faststore.dev/how-to-guides/troubleshooting/analytics-and-partytown#see-events-pushed-to-datalayer):

Before|After
-|-
![CleanShot 2022-09-13 at 17 37 03](https://user-images.githubusercontent.com/381395/190004361-0827bb1b-139c-492a-a7d7-1017e634b69d.png)|![CleanShot 2022-09-13 at 17 36 06](https://user-images.githubusercontent.com/381395/190004382-0790588a-f1a7-4e94-8ece-34f3b723e796.png)
![CleanShot 2022-09-13 at 17 33 52](https://user-images.githubusercontent.com/381395/190003810-58db1994-ed26-49d4-8aec-cd7ce38b2a3e.png)|![CleanShot 2022-09-13 at 17 33 29](https://user-images.githubusercontent.com/381395/190003824-4a71cdd6-0b93-4b73-a29b-bfdfb3240e84.png)


## References

- [[Internal] Jira task FS-469](https://vtex-dev.atlassian.net/browse/FS-469)
- https://www.faststore.dev/how-to-guides/troubleshooting/analytics-and-partytown#see-events-pushed-to-datalayer for explaining how to check events being pushed to the `dataLayer`.
